### ISSUE TITLE
Textfield password alias not working

### ISSUE THREAD

**Issue by jcteague** [xinthink/react-native-material-kit#220]
We had a password text field defined as so:

```JavaScript
const PasswordField = MKTextField
  .textfieldWithFloatingLabel()
  .withPassword(true)
  .withPlaceholder('Password:')
  .withStyle(styles.textfieldWithFloatingLabel)
  .withFloatingLabelFont({fontSize: 10, fontStyle: 'italic', fontWeight: '200'})
  .withTintColor(MKColor.Red)
  .build();
```
After upgrading to r-n .30 from .24, it was no longer hiding the password characters.  We did a workaround by specifying the actual proptype name on textfield `secureTextEntry` and the characters are now masked.

```JavaScript
const PasswordField = MKTextField
  .textfieldWithFloatingLabel()
  //works with correct property name, not with alias
  .withSecureTextEntry(true)
  .withPlaceholder('Password:')
  .withStyle(styles.textfieldWithFloatingLabel)
  .withFloatingLabelFont({fontSize: 10, fontStyle: 'italic', fontWeight: '200'})
  .withTintColor(MKColor.Red)
  .build();
```

**Comment by jcteague** [comment]
I just tested to see if the `value` alias in textfield would work and it is also not working.

I was going to work on PR for this, but I honestly don't think these provide any real value.  IMHO just changing the docs would be better.  It keeps the builder props inline with r-n props and you don't have to make any changes if they change property names.

**Comment by Crash--** [comment]
Thank you @jcteague! I'll test it and see if it works well everywhere and then update the readme.

**Comment by jcteague** [comment]
I should have mentioned I only tested on android

**Comment by homanp** [comment]
How do I set the password type in MKTextField using JSX? I've tried both `password` and `secureTextEntry` as props, none of them seem to work.

Example below:
`<MKTextField
            highlightColor={'#00C84B'}
            placeholder='Password'
            style={styles.textfieldWithFloatingLabel}
            floatingLabelEnabled={true}
            keyboardType={'email-address'}
            value={this.state.loginData.password}
            password={true}
            onTextChange={(password) => this.setState({password})}
          />`

**Comment by jcteague** [comment]
@homanp this worked just fine for me on android.  I'm not setup on ios at the moment, so I can't test there.
```jsx
          <MKTextField
            highlightColor={'#00C84B'}
            placeholder='Password' style={style.textfieldWithFloatingLabel}
            floatingLabelEnabled={true}

            secureTextEntry={true}
            onTextChange={(password) => this.setState({password})} />
```

**Comment by homanp** [comment]
Hmm...I'm running RN 0.27. Perhaps that is an issue?

**Comment by homanp** [comment]
The above code doesn't mask the password field input

**Comment by jcteague** [comment]
could be, I tested on .30

**Comment by homanp** [comment]
The 0.27 states that secureTextEntry should work on both iOS and Android. Any ideas on what could be wrong?

**Comment by homanp** [comment]
Btw tested with te build() merhod and withSecureTextEntry works just fine

**Comment by Crash--** [comment]
Yep it doesn't work with JSX. There is a PR (if I remember correctly) to fix it.. I need time to merge it.

**Comment by vermiculite** [comment]
There is a pull request to fix it. It is just a one line fix. If I can do anything to help let me know and I will do it.

**Comment by CFitzsimons** [comment]
I'm still having this issue, not sure if it's something I've done wrong or not but thought I'd provide more information.

**I am only running on iOS**

RN: 0.31.0
CLI: 0.2.0
RNMK: 0.3.4
iOS Simulator: iPhone 5s
iOS Version: 9.3

I've tested with a builder:

```
const MaterialPasswordField = MKTextField.textfield()
  .withSecureTextEntry(true)
  .withStyle(Style.iOSRequired)
  .withTextInputStyle(Style.inputStyle)
  .withHighlightColor(MKColor.Lime)
  .withTintColor(MKColor.Blue)
  .withPlaceholderTextColor(Colors.placeholderColor)
  .build();
```

and without:

```
<MKTextField 
    testID="textfield_password" 
    secureTextEntry={true} 
    returnKeyType="done" 
    autoCapitalize={'none'} 
    autoCorrect={false} 
    onChangeText={(text) => { this.setState({ password: text }); }} 
  />
```

Sorry if my styling/formatting isn't great.

**Comment by damoguyan8844** [comment]
use <MKTextField
              ....
              password={true}
            /> works
